clash-lib/src/Clash/Driver.hs

```diff
@@ -214,6 +214,8 @@ generateHDL reprs bindingsMap hdlState primMap tcm tupTcm typeTrans eval
     go prepTime HashMap.empty (sortTop bindingsMap topEntities1)
  where
   topEntities1 = map (splitTopEntityT tcm bindingsMap) topEntities0
+  -- Remove inline type equality constraints
+  topEntities = map (\(TopEntityT var annM tbM) -> TopEntityT var{varType=tvSubstWithTyEq (varType var)} annM tbM) topEntities1
```

Comment:
It's probably best to call this `topEntities2` to clarify that it's the "descendant" of `topEntities1`.

Reply:
done